Team1CodeForReview

CedricKapalla opened the pull request:
Texteditor Implementierung mit generellen editor-tools (open, save, copy/paste/cut, etc.) fertig implementiert, mehreren tabs, öffnung und schließung von tabs, und word-/linecount.

txt, java, und xml Dateien können per Doppelklick vom internen file-explorer aus geöffnet werden.

Hat noch einige kleinere bugs (wordcount funktioniert nicht direkt nach dem öffnen, erst wenn ein zweiter Tab geöffnet wurde; öffnen eines tabs setzt nicht direkt den cursor), sollte aber ansonsten funktionieren.

lukcron committed:
Basis commit

lukcron committed:
Added eclipse project files to .gitignore

erwin97 committed:
Test push

SoerenC96 committed:
Ordnerstruktur angepasst

SoerenC96 committed:
Merge branch 'master' of https://github.com/Narorororo/e4CFTeam1WS20-21.git

SoerenC96 committed:
Anpassung an neue Struktur

SoerenC96 committed:
Erste Menuitems +Platzhalter

SoerenC96 committed:
Erste Buttons + Platzhalter

CedricKapalla committed:
Initialisation for remaining menu items, placeholder messages; no commentary yet, will follow soon

CedricKapalla committed:
Comments for most of the menu items and functions.

CedricKapalla committed:
Added functionality to menu items: Cut, Copy, Paste.

erwin97 committed:
Implementation of Open,Save,SaveAs,New

erwin97 committed:
Implementation of New,Open,Save,SaveAs

lukas410 committed:
Merge commit

lukas410 committed:
Changed string comparision in FileUtils to be based on hashCode

lukas410 committed:
Fixed return in FileUtils save method

lukas410 committed:
Added more comments in TextEditor

CedricKapalla committed:
Merge branch 'master' of https://github.com/CedricKapalla/e4CFTeam1WS20-21

# Conflicts:
#	de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/view/TextEditor.java

CedricKapalla committed:
Merge branch 'master' of https://github.com/CedricKapalla/e4CFTeam1WS20-21

lukas410 committed:
Added comments in FileUtils

CedricKapalla committed:
Merge branch 'master' of https://github.com/CedricKapalla/e4CFTeam1WS20-21

erwin97 committed:
Adding popup error message if save failed & add selectAll Method

erwin97 committed:
adding gitignore

SoerenC96 committed:
Merge Commit

SoerenC96 committed:
Merge Commit

CedricKapalla committed:
Added functions undo, redo

lukcron committed:
Made TextEditor closable and set workspace dir as initial directory

SoerenC96 committed:
Label angepasst

CedricKapalla committed:
Cleared errors with wordCount and rowCount, comments still missing.

CedricKapalla committed:
Added 'save file' dialogs for New File-action, Close File-action, Close Editor-action.

CedricKapalla committed:
Added alert windows as placeholders for The Preferences-menu Item and About-menu Item.

CedricKapalla committed:
Quick hotfix, missed a placeholder-line to delete

SoerenC96 committed:
Kommentare hinzugefügt

SoerenC96 committed:
Kommentare hinzugefügt

erwin97 committed:
FileExtension

lukas410 committed:
Refactored JavaFileExtension

SoerenC96 committed:
Save angepasst

lukas410 committed:
Refactored FileUtils

SoerenC96 committed:
Event works now

CedricKapalla committed:
Event handling implemented for all extension points (txt, java, xml), currently placeholders

CedricKapalla committed:
Finished implementing "close editor" menu item.

CedricKapalla committed:
Added documentation, moved supporting functions in TextEditor to the bottom

SoerenC96 committed:
Tabulator und andere Zählprobleme beseitigt

SoerenC96 committed:
StringTable hinzugefügt

SoerenC96 committed:
TabPane hinzugefügt + Dateien über Project Explorer öffnen

erwin97 committed:
CodeArea verwenden + Funktionalität anpassen + Tabs implementieren

lukas410 committed:
Added helper methods to TextEditor.java

lukas410 committed:
Removed accidental .project file

lukas410 committed:
Removed correct .project file

SoerenC96 committed:
Bug Fixing + Verbesserungen

SoerenC96 committed:
Anfanh vom Focus (focus request funktioniert nicht)

lukcron committed:
Made files always open in new Tab, Added linenumbers

lukcron committed:
Fixed editor close event

lukcron committed:
Added shortcuts to menu

lukcron committed:
Added filename loading for saveAs

CedricKapalla committed:
Refactored function parseFileNameFromPath to FileUtils, fixed issue of .txt file opening throwing errors when including umlauts

SoerenC96 committed:
1. Placeholde for icons

2. One general fileextension

3. Count Bug

4. Renamed to extension

SoerenC96 committed:
1. New angepasst für das Highlighting

CedricKapalla committed:
Added documentation, added icon, fixed file opening bug that did not read file ending properly, implemented custom Tab class EditorTab.

CedricKapalla committed:
Added EditorTab and Icon to repository

erwin97 committed:
Apply Syntax Highlighting

erwin97 committed:
Merge branch 'master' of https://github.com/CedricKapalla/e4CFTeam1WS20-21.git

erwin97 committed:
Syntax Highlighting without comment

lukcron committed:
Refactored SyntaxHighlighter

lukcron committed:
Improved XML Highlighting

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/TextEditorViewController.java:
```diff
@@ -0,0 +1,32 @@
+import javafx.scene.Scene;
+
+public class TextEditorViewController {
+	private static final String TEXT_EDITOR_VIEW_FXML = "TextEditorView.fxml";
```
Formatierung

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/TextEditorViewController.java:
```diff
@@ -0,0 +1,32 @@
+import javafx.embed.swt.FXCanvas;
+import javafx.scene.Scene;
+
+public class TextEditorViewController {
```
Java Doc

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/view/TextEditor.java:
```diff
@@ -0,0 +1,5 @@
+package de.tu_bs.cs.isf.e4cf.text_editor.view;
+
```
Java Doc

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,204 @@
+import de.tu_bs.cs.isf.e4cf.core.util.RCPMessageProvider;
+import de.tu_bs.cs.isf.e4cf.core.util.ServiceContainer;
+import de.tu_bs.cs.isf.e4cf.text_editor.stringtable.EditorST;
+
```
Java Doc

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,204 @@
+	private int lastSavedRevision;
+
+	@Inject
+	private ServiceContainer services;
```
Das funktioniert glaube ich nicht weil sich die Klasse nicht im Kontext befindet

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,204 @@
+				splittedPath = path.split("\\\\");
+			} else {
+				splittedPath = path.split("/");
+			}
```
strg + shit +f  also auto formatierung mal auf allem drücken

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,204 @@
+	/**
+	 * Opens the file chosen by the open dialog.
+	 *
+	 * @return String[] of length 2 with (absolute) filePath at index 0 and
```
was ist index 1 ?

timbittner replied:
probably nur doc typo, file-content at index 1

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/highlighter/JavaHighlighting.java:
```diff
@@ -0,0 +1,72 @@
+
+import org.fxmisc.richtext.model.StyleSpans;
+import org.fxmisc.richtext.model.StyleSpansBuilder;
+
```
JavaDoc

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/extensions/ProjectExplorerFileExtension.java:
```diff
@@ -0,0 +1,29 @@
+	@Override
+	public Image getIcon(RCPImageService imageService) {
+		// TODO Auto-generated method stub
+		return null;
```
error :P

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/highlighter/XMLHighlighting.java:
```diff
@@ -0,0 +1,82 @@
+
+import org.fxmisc.richtext.model.StyleSpans;
+import org.fxmisc.richtext.model.StyleSpansBuilder;
+
```
JavaDoc

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/view/EditorTab.java:
```diff
@@ -0,0 +1,49 @@
+	}
+
+	/**
+	 * Creates a new CodeArea and adds text to it, if any is given.
```
public static methoden sind immer ein guter hinweis auf ich extrahiere es in eine factory / builder / util Klasse

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/view/TextEditor.java:
```diff
@@ -0,0 +1,661 @@
+ * 
+ * @author Soeren Christmann, Cedric Kapalla, Lukas Cronauer, Erwin Wijaya
+ *
+ */
```
Die Klasse ist halt echt ziemlich groß. Ich denke, ihr solltet eure View in Mehren FXML Files mit jeweiligen Controllern oder so schreiben damit das alles etwas übersichtlicher wird, bzw. ich weiß nicht, ob das so geht aber ein Versuch ist es sicherlich wert.

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/view/TextEditor.java:
```diff
@@ -0,0 +1,661 @@
+	 * in another file directory Or make a copy of the file with another name
+	 * 
+	 * @author Lukas Cronauer, Erwin Wijaya
+	 */
```
Das kann man definitive schöner machen mit FXML spart pro methode locker 5-6 Zeilen.
onAction="#methodenName" im XML und dann mit @FXML die Methode annotieren.

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/view/TextEditor.java:
```diff
@@ -0,0 +1,661 @@
+		});
+	}
+
+	/**
```
Dann braucht man auch diese Methoden alle nicht mehr

KamilRosiak reviewed (changes requested):
Formatiert mal den Code mit Strg-Shift +O und guckt mal, ob man die View irgendwie Modularisieren kann. Ich finde, der Controller ist schon ganz schön groß. 

Sonst sieht das schon recht gut aus.

Jooshua commented:
Hey team01, wir werden euren PR reviewen nachdem ihr die Changes von @KamilRosiak eingebaut habt :)

Eine Sache die ich euch allgemein ans Herz legen möchte:
Ihr könnt auf Github Commits (/PRs) einem Ticket zuordnen, indem ihr in der Commit-Message die Nummer des Tickets mit #Nummer (z.B. #29) referenziert. Gleichzeitig das könnt ihr das auch bei PRs auf andere Repos (wie bei diesem) über die URL (z.B. https://github.com/CedricKapalla/e4CFTeam1WS20-21/issues/6). 
Mehr unter: https://guides.github.com/features/issues/

Das macht das ganze nachvollziehbarer und einfacher zu reviewen ;)

lukcron committed:
Added first fxml method

CedricKapalla committed:
Refining of menu item actions #6

-Adjusted all menu items to work on actions defined in fxml file.
-Removed most initialisers, which were no longer necessary for the items to work.
-Adjusted initialize function for new setup (removed other initialisers)
-Moved createNewFileItems from bottom of TextEditor class to top.
-Renamed getCurrentTextArea to getCurrentCodeArea (previous update changed data type used).
-renamed all variables "textArea" to "codeArea"

SoerenC96 committed:
Annotations Highlighting #5

SoerenC96 committed:
Annotation Bug Fix #5

erwin97 committed:
Added Comment

erwin97 committed:
Count method in new class, adding new regex for instance (javaHighlight)

erwin97 committed:
divided fxml into module

timbittner commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,203 @@
+	 * @return String the text from the file
+	 * @author Lukas Cronauer, Erwin Wijaya
+	 */
+	public String readFile(File file) {
```
Method throws NPE and is not declared throwing.
Also handles other exceptions, seems inconsistent

timbittner commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,203 @@
+		if (f != null && writeFile(f.getAbsolutePath(), content)) {
+			return f.getAbsolutePath();
+		} else {
+			return "";
```
From javadoc this method should always return some kind of path.
Is the else branch some kind of error handling?

timbittner commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/highlighter/SyntaxHighlighter.java:
```diff
@@ -0,0 +1,167 @@
+     * @param fileType extension of the file
+     * @param codeArea current area for text on active tab
+     */
+    private void initHighlighting(String fileType, CodeArea codeArea) {
```
Just put these two calls into the constructor?

timbittner commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/highlighter/SyntaxHighlighter.java:
```diff
@@ -0,0 +1,167 @@
+	            break;
+	
+	        default:
+	        	return;
```
`break` instead of `return `for void function

timbittner commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/highlighter/SyntaxHighlighter.java:
```diff
@@ -0,0 +1,167 @@
+	 * @param codeArea current area for text on active tab
+	 */
+    private void scheduleHighlighting(CodeArea codeArea) {
+//         executor = Executors.newSingleThreadExecutor();
```
Is there a reason for this commented out code?

erwin97 committed:
added some function for javaHighlighting, it will now highlight instance

erwin97 committed:
Fixing bug, where sometimes System.out.print will be highlighted. Found
another bug: JavaNullPointerException by trying to opening .txt file

lukcron committed:
Created NoHighlighting class

lukcron committed:
Created factory class for CodeArea

lukcron committed:
Added JavaDoc-Comments, Auto-formatted all classes

SoerenC96 committed:
Icons

lukcron committed:
Created three separate ProjectExplorerFileExtensions

lukcron committed:
Merge branch 'master' of https://github.com/Narorororo/e4CFTeam1WS20-21.git

lukcron committed:
Renamed icons, Added icons in fileExtensions

lukcron committed:
Deleted ProjectExplorerFileExtension

lukcron committed:
Prevented IllegalStateExceptions when no highlighting match is found

lukcron committed:
Wrote JavaDoc-Comments

kyshidar commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/highlighter/SyntaxHighlighter.java:
```diff
@@ -0,0 +1,165 @@
+	 *
+	 * @author Lukas Cronauer
+	 */
+	public void canceHighlighting() {
```
cancelHighlighting() misses the "l"

kyshidar commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/highlighter/SyntaxHighlighter.java:
```diff
@@ -0,0 +1,165 @@
+		String text = codeArea.getText();
+		Task<StyleSpans<Collection<String>>> task = new Task<StyleSpans<Collection<String>>>() {
+			@Override
+			protected StyleSpans<Collection<String>> call() throws Exception {
```
missing exception handling - could stop the task execution

valentinpurrucker commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,211 @@
+					throw new ArrayIndexOutOfBoundsException("Invalid filename");
+				}
+			} catch (ArrayIndexOutOfBoundsException e) {
+				System.out.println(e.getMessage());
```
Maybe dispatch the error message to the messageprovider if necessary at all or remove the print statement.

Jooshua reviewed (approved):
No additional changes from my side :)

valentinpurrucker commented on de.tu_bs.cs.isf.e4cf.text_editor/src/de/tu_bs/cs/isf/e4cf/text_editor/FileUtils.java:
```diff
@@ -0,0 +1,211 @@
+		try {
+			reader = new FileReader(file);
+			int character;
+			String text = "";
+
+			while ((character = reader.read()) != -1) {
+				text += (char) character;
+			}
```
Maybe you could use a StringBuilder to avoid unnecessary object allocation.

valentinpurrucker reviewed (approved):
Only short feedback that you could look into. Changes are not mandatory for approval.
